src/mon/MgrMonitor.cc

```diff
@@ -70,63 +70,33 @@ static ostream& _prefix(std::ostream *_dout, Monitor &mon,
 
 // the system treats always_on_modules as if they provide built-in functionality
 // by ensuring that they are always enabled.
-const static std::map<uint32_t, std::set<std::string>> always_on_modules = {
-  {
-    CEPH_RELEASE_OCTOPUS, {
-      "crash",
-      "status",
-      "progress",
-      "balancer",
-      "devicehealth",
-      "orchestrator",
-      "rbd_support",
-      "volumes",
-      "pg_autoscaler",
-      "telemetry",
-    }
-  },
-  {
-    CEPH_RELEASE_PACIFIC, {
-      "crash",
-      "status",
-      "progress",
-      "balancer",
-      "devicehealth",
-      "orchestrator",
-      "rbd_support",
-      "volumes",
-      "pg_autoscaler",
-      "telemetry",
-    }
-  },
-  {
-    CEPH_RELEASE_QUINCY, {
-      "crash",
-      "status",
-      "progress",
-      "balancer",
-      "devicehealth",
-      "orchestrator",
-      "rbd_support",
-      "volumes",
-      "pg_autoscaler",
-      "telemetry",
-    }
-  },
-  {
-    CEPH_RELEASE_REEF, {
+static const std::map<uint32_t, std::set<std::string>>& release_always_on_modules() {
+  static std::map<uint32_t, std::set<std::string>> always_on_modules;
+  static std::once_flag did_init;
+  std::call_once(did_init, []() {
+    std::set<std::string> octopus_modules = {
       "crash",
       "status",
       "progress",
       "balancer",
       "devicehealth",
       "orchestrator",
-      "rbd_support",
-      "volumes",
       "pg_autoscaler",
       "telemetry",
-    }
-  },
+    };
+#   ifdef WITH_CEPHFS
+      octopus_modules.insert("volumes");
+#   endif
```

Comment:
Nit: don't indent `ifdef` itself and don't indent _based on_ `ifdef`.  This would be better formatted as:

```suggestion
#ifdef WITH_CEPHFS
    octopus_modules.insert("volumes");
#endif
```
One (or their editor/IDE) should be able to hide preprocessor directives (picking one of the branches) and not have to reindent everything.